### PR TITLE
Add Image Container

### PR DESCRIPTION
**Pull request by jtart**
The Image container component renders a Figure presentational component, with optional Caption presentational component. PropTypes are added to Figure and Caption. PropTypes for the  container component come from the cps model. 

* [x] Partial fix for https://github.com/bbc/simorgh/issues/246
* [x] Tests added for new features

* [ ] Test engineer approval

### CODE REVIEW

**Comment by radiocontrolled** [comment]
How come there isn't a standalone Figure component? @jtart @mattgreenham

File: src/app/components/Figure/Caption/index.jsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import styled from 'styled-components';
+import { FF_NEWS_SANS_REG } from '../../../lib/constants/styles';
+
+const Caption = styled.figcaption`
+  background-color: #d5d0cd;
+  color: #404040;
```
Comment (pjlee11) [inline src/app/components/Figure/Caption/index.jsx]: This should be `${C_STORM}` https://github.com/bbc/simorgh/blob/data-schema-validator-part-2/src/app/lib/constants/styles.js#L7

File: src/app/components/Figure/Caption/index.jsx [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import styled from 'styled-components';
+import { FF_NEWS_SANS_REG } from '../../../lib/constants/styles';
+
+const Caption = styled.figcaption`
+  background-color: #d5d0cd;
```
Comment (pjlee11) [inline src/app/components/Figure/Caption/index.jsx]: Opening an issue to confirm this value with UX and add it to the constant style values and then reuse it against the caption and image component.

File: src/app/components/Figure/index.jsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import { string } from 'prop-types';
+import Caption from './Caption';
+
+function renderCaption(caption) {
```
Comment (pjlee11) [inline src/app/components/Figure/index.jsx]: Is there a reason this isn't in ES6 syntax? 

EG:
```
const renderCaption = caption =>
  caption ? <Caption>{caption}</Caption> : null;
```

File: src/app/components/Figure/index.stories.jsx [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+import Figure from './index';
+
+const image = {
+  alt:
+    'Map of the UK displaying Syrian refugees and asylum seekers per 10000 population. Ranges from 0 to 17.',
+  src:
+    'https://ichef.bbci.co.uk/news/640/cpsprodpb/439A/production/_100960371_syrians_and_asylum_v2-nc.png',
+};
+
+const caption = 'This is a caption!';
```
Comment (pjlee11) [inline src/app/components/Figure/index.stories.jsx:12]: Can we add a story without a caption to show the variation of the figure component

**Review by pjlee11 — CHANGES_REQUESTED** [review]
Couple of points to address

File: src/app/components/Figure/Caption/index.jsx (hunk 1 above)
Comment (pjlee11) [reply src/app/components/Figure/Caption/index.jsx]: https://github.com/bbc/simorgh/issues/322

File: src/app/components/Figure/Caption/index.jsx (hunk 2 above)
Comment (pjlee11) [reply src/app/components/Figure/Caption/index.jsx]: Opened issue https://github.com/bbc/simorgh/issues/322 to address this

File: src/app/components/Figure/index.stories.jsx [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+storiesOf('Figure', module)
+  .add('with a caption', () => <Figure {...image} caption={caption} />)
+  .add('without a caption', () => <Figure {...image} />);
```
Comment (radiocontrolled) [inline src/app/components/Figure/index.stories.jsx]: @pjlee11 does this not address your comment above?

File: src/app/components/Figure/index.stories.jsx (hunk 4 above)
Comment (radiocontrolled) [reply src/app/components/Figure/index.stories.jsx:12]: This is addressed in https://github.com/bbc/simorgh/pull/296/files#diff-29479ec42fe7d690788103a2bc221ba7R16

**Review by pjlee11 — DISMISSED** [review]
Great work

File: src/app/containers/Image/index.stories.jsx [hunk 6]
```diff
@@ -38,9 +38,10 @@ const dataWithCaption = imageData([
   ),
 ]);
 
-storiesOf('Image', module).add('image with alt text', () => (
-  <Image {...dataWithAltText} />
-));
-storiesOf('Image', module).add('image with alt text and caption', () => (
-  <Image {...dataWithCaption} />
+storiesOf('ImageContainer', module).add('image with alt text', () => (
+  <ImageContainer {...dataWithAltText} />
```
Comment (jamesbhobbs) [inline src/app/containers/Image/index.stories.jsx]: can we just call this data or dataWithoutCaption the inconsistent naming between this and below (which doesn't specify that it has alt text is confusing.

Comment (jamesbhobbs) [reply src/app/containers/Image/index.stories.jsx]: same in the test file

**Review by jamesbhobbs — CHANGES_REQUESTED** [review]
Trivial changes otherwise mergeable

**Review by jamesbhobbs — APPROVED** [review]

**Pull request merged**